extensions/wikia/CommunityHeader/Navigation.class.php

```diff
@@ -2,93 +2,56 @@
 
 namespace Wikia\CommunityHeader;
 
-use \NavigationModel;
-use \Title;
+use DesignSystemCommunityHeaderModel;
+use NavigationModel;
 
 class Navigation {
 	public $discussLink;
 	public $exploreItems;
 	public $exploreLabel;
 	public $localNavigation;
 
-	public function __construct( $wikiText = null ) {
-		$navigationModel = new NavigationModel();
-		if ( empty( $wikiText ) ) {
-			$this->localNavigation =
-				$navigationModel->getLocalNavigationTree( NavigationModel::WIKI_LOCAL_MESSAGE );
-		} else {
-			$this->localNavigation = $navigationModel->getTreeFromText( $wikiText );
-		}
+	private $model;
+
+	public function __construct( DesignSystemCommunityHeaderModel $model, $wikiText = null ) {
+		$this->model = $model;
+
+		$this->localNavigation = $model->getWikiLocalNavigation( $wikiText );
+
+		$exploreMenu = $this->model->getExploreMenu();
```

Comment:
Nitpick: this can be `$model->getExploreMenu()` and I would remove the empty lines between 3 (5) lines above.